Cannot build locally because of unavailable dependency org.simplejavamail:utils-mail-dkim:3.2.1-haumacher

**Is**

When trying to build PhoneBlock locally using `mvn compile`, I am getting an error:

```bash
$ mvn compile
[...]
[INFO] ----------------------< de.haumacher:phoneblock >-----------------------
[INFO] Building PhoneBlock 1.7.8-SNAPSHOT                                 [4/4]
[INFO]   from phoneblock\pom.xml
[INFO] --------------------------------[ war ]---------------------------------
[WARNING] The POM for org.simplejavamail:utils-mail-dkim:jar:3.2.1-haumacher is missing, no dependency information available
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for PhoneBlock Aggregator 1.7.8-SNAPSHOT:
[INFO] 
[INFO] PhoneBlock Aggregator .............................. SUCCESS [  0.661 s]
[INFO] PhoneBlock shared code ............................. SUCCESS [  2.625 s]
[INFO] PhoneBlock Anrufbeantworter ........................ SUCCESS [  1.605 s]
[INFO] PhoneBlock ......................................... FAILURE [  1.197 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  6.252 s
[INFO] Finished at: 2025-09-25T11:40:57+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project phoneblock: Could not resolve dependencies for project de.haumacher:phoneblock:war:1.7.8-SNAPSHOT
[ERROR] dependency: org.simplejavamail:utils-mail-dkim:jar:3.2.1-haumacher (compile)
[ERROR]         org.simplejavamail:utils-mail-dkim:jar:3.2.1-haumacher was not found in https://repo.maven.apache.org/maven2 during a previous attempt. This failure was cached in the local repository and resolution is not reattempted until the update interval of central has elapsed or updates are forced                                                                                                                                                                                                                                                                                          
[ERROR] 
```

This is related to the dependency org.simplejavamail:utils-mail-dkim:3.2.1-haumacher in https://github.com/haumacher/phoneblock/blob/e8d1adc91e15c1003e1d5e793785e3f27ce66adb/phoneblock/pom.xml#L286

**Please fix**

I assume org.simplejavamail:utils-mail-dkim:3.2.1-haumacher exists because of packaged credentials.

If possible, please change the signing mechanism so others can build this project locally, too.

--- a/phoneblock-ab/pom.xml
+++ b/phoneblock-ab/pom.xml
@@ -46,18 +46,17 @@
 		    <version>${project.version}</version>
 		</dependency>
 		
-		<dependency>	
-		    <groupId>org.mjsip</groupId>
+		<dependency>
+		    <groupId>com.github.haumacher.mjSIP</groupId>
 			<artifactId>mjsip-ua</artifactId>
 		    <version>2.0.5</version>
 		</dependency>
 	</dependencies>
-	
+
 	<repositories>
 		<repository>
-			<id>github</id>
-			<name>GitHub haumacher/mjSIP Apache Maven Packages</name>
-			<url>https://maven.pkg.github.com/haumacher/mjSIP</url>
+			<id>jitpack.io</id>
+			<url>https://jitpack.io</url>
 			<releases>
 				<enabled>true</enabled>
 			</releases>

--- a/phoneblock/pom.xml
+++ b/phoneblock/pom.xml
@@ -300,10 +300,6 @@
 	</dependencies>
 
 	<build>
-		<filters>
-			<filter>.phoneblock</filter>
-		</filters>
-	
 		<resources>
 			<resource>
 				<directory>${project.build.sourceDirectory}</directory>
@@ -313,7 +309,6 @@
 			</resource>
 			<resource>
 				<directory>${project.basedir}/src/main/resources-filtered</directory>
-				<filtering>true</filtering>
 			</resource>
 		</resources>
 

--- a/phoneblock/pom.xml
+++ b/phoneblock/pom.xml
@@ -300,6 +300,10 @@
 	</dependencies>
 
 	<build>
+		<filters>
+			<filter>.phoneblock</filter>
+		</filters>
+
 		<resources>
 			<resource>
 				<directory>${project.build.sourceDirectory}</directory>
@@ -309,10 +313,39 @@
 			</resource>
 			<resource>
 				<directory>${project.basedir}/src/main/resources-filtered</directory>
+				<filtering>true</filtering>
 			</resource>
 		</resources>
 
 		<plugins>
+			<plugin>
+				<groupId>org.apache.maven.plugins</groupId>
+				<artifactId>maven-enforcer-plugin</artifactId>
+				<executions>
+					<execution>
+						<id>check-phoneblock-config</id>
+						<phase>validate</phase>
+						<goals>
+							<goal>enforce</goal>
+						</goals>
+						<configuration>
+							<rules>
+								<requireFilesExist>
+									<files>
+										<file>${project.basedir}/.phoneblock</file>
+									</files>
+									<message>
+Configuration file '.phoneblock' is missing.
+Copy '.phoneblock.template' to '.phoneblock' and fill in your local values
+(database URL, OAuth credentials, SMTP settings). See CLAUDE.md for details.
+									</message>
+								</requireFilesExist>
+							</rules>
+						</configuration>
+					</execution>
+				</executions>
+			</plugin>
+
 			<plugin>
 				<groupId>de.haumacher.msgbuf</groupId>
 				<version>${msgbuf.version}</version>

